core/src/main/scala/org/apache/spark/h2o/converters/LabeledPointConverter.scala

```diff
@@ -46,9 +45,17 @@ private[converters] object LabeledPointConverter extends Logging with ConverterU
       logWarning("WARNING: Converting RDD[LabeledPoint] to H2OFrame where features vectors have different size, filling missing with n/a")
     }
     val fnames = ("label" :: 0.until(maxNumFeatures).map("feature" +).toList).toArray[String]
-    val vecTypes = Array.fill(maxNumFeatures + 1)(Vec.T_NUM)
 
-    convert[LabeledPoint](hc, rdd, keyName, fnames, vecTypes, perLabeledPointRDDPartition(maxNumFeatures))
+    // in case of internal backend, store regular vector types
+    // otherwise for external backend store expected types
+    val expectedTypes = if(hc.getConf.runsInInternalClusterMode){
+      Array.fill(maxNumFeatures + 1)(Vec.T_NUM)
+    }else{
+      Array.fill(maxNumFeatures + 1)(ExternalFrameUtils.EXPECTED_DOUBLE)
+    }
```

Comment:
Yup, it make sense in this case. I was putting this piece of code amongst all the converters, but at this particular one the simplification make sense 👍